packages/docusaurus/src/server/routes.ts

```diff
@@ -66,7 +66,11 @@ ${indent(removeSuffix(subroutesCodeStrings.join(',\n'), ',\n'))}
     // Is this lib reliable? https://github.com/armanozak/should-quote
     const shouldQuote = true; // TODO
     const key = shouldQuote ? `'${propName}'` : propName;
-    parts.push(`${key}: ${JSON.stringify(propValue)}`);
+    const value =
+      typeof propValue === 'function'
+        ? propValue.toString()
```

Comment:
I don't like this API. It breaks the isolation of client-server and means plugin authors can write arbitrary JS code that's executed on the client side. This has never been the case before, and I don't feel like it's good.